test/jdk/sun/security/x509/X500Name/PrintX500NameInDebugModeWithAvaOption.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/* @test
+ * @bug 8349890
+ * @summary Make sure debug with AVA option does not interfere with parsing special characters.
+ * @modules java.base/sun.security.x509
+ * @library /test/lib
+ * @run main/othervm -Djava.security.debug=x509:ava PrintX500NameInDebugModeWithAvaOption
+ */
+
+import jdk.test.lib.Asserts;
+import sun.security.x509.X500Name;
+
+public class PrintX500NameInDebugModeWithAvaOption {
+
+    public static void main(String[] args) throws Exception {
+
+        X500Name name = new X500Name("cn=john doe + l=ca\\+lifornia + l =sf, O=Ñ");
+
+        //Test the name in default String format. This will perform the hex conversion to
+        //"\\C3\\91" for special character "Ñ"
+        Asserts.assertTrue(name.toString().contains("\\C3\\91"),"String does not contain expected value");
+
+        //Test the name in RFC2253 format. This should skip the hex conversion to return
+        //"\u00d1" for special character "Ñ"
+        Asserts.assertTrue(name.getRFC2253Name().contains("\u00d1"), "String does not contain expected value");
+
+        //Test the name in canonical name in RFC2253 format. This should skip the hex conversion to return
+        //"n\u0303" for special character "Ñ"
+        Asserts.assertTrue(name.getRFC2253CanonicalName().contains("n\u0303"),"String does not contain expected value");
```

Comment:
Here you can call `X500Principal.getName(X500Principal.CANONICAL)` instead which emits a canonical RFC 2253 name.